core/src/main/scala/org/apache/spark/util/ThreadUtils.scala

```diff
@@ -156,4 +156,11 @@ private[spark] object ThreadUtils {
         result
     }
   }
+
+  /**
+   * Construct a new ForkJoinPool with a specified max parallelism.
+   */
+  def newForkJoinPool(maxThreadNumber: Int): scala.concurrent.forkjoin.ForkJoinPool = {
+    new scala.concurrent.forkjoin.ForkJoinPool(maxThreadNumber)
```

Comment:
So, with the scala ForkJoinPool even when provide a thread factory, it sets the thread names based on the "workerNamePrefix" (which is final) inside of "registerWorker" which is called when constructing a new ForkJoinWorkerThread - although with a custom factory we can change the name back after the thread has been constructed.